24/Service registry

tahpot opened the pull request:
Closes #24 .

Initial MVP of service registry.

ITStar10 committed:
Updated for gasles transaction server

Remaining:
	node-test/getlogs_*.js

ITStar10 committed:
Updated for gasles transaction server

Remaining:
	node-test/getlogs_*.js

ITStar10 committed:
Minor updte for Package.json

ITStar10 committed:
Minor updte for Package.json

ITStar10 committed:
Updates by Slither check

BytesLib.sol
	Removed unnecessary functions
VeridaDIDRegistry.sol
	Update function modifiers from 'public' to 'external'

ITStar10 committed:
Updates by Slither check

BytesLib.sol
	Removed unnecessary functions
VeridaDIDRegistry.sol
	Update function modifiers from 'public' to 'external'

BlockchainCrazy95 committed:
Initial commit

BlockchainCrazy95 committed:
Initial commit

BlockchainCrazy95 committed:
Comment Added

- Add comments in the body of functions
- Add discoverService() function

BlockchainCrazy95 committed:
Comment Added

- Add comments in the body of functions
- Add discoverService() function

BlockchainCrazy95 committed:
Minor updates

BlockchainCrazy95 committed:
Minor updates

BlockchainCrazy95 committed:
Update

BlockchainCrazy95 committed:
Update

ITStar10 committed:
Updated visibility of functions

Update public methods to external

ITStar10 committed:
Updates on Review

Added comment in NatSpec
Updated configuration : using .env

ITStar10 committed:
Updates on Review

Added comment in NatSpec
Updated configuration : using .env

pranavburnwal committed:
Merge pull request #30 from verida/Alex/VeridaToken-Update

ITStar10 committed:
Minor updates for hardhatconfig.ts

ITStar10 committed:
Merge branch 'feature/10-did-registry' of github.com:verida/blockchain-contracts into feature/10-did-registry

pranavburnwal committed:
Merge pull request #27 from verida/feature/10-did-registry

ITStar10 reviewed (commented):
- Need updates for credit. Or it can be resolved while implementing treasury.
- Need service status check in functions

ITStar10 reviewed (commented):
- Consolidate unnecessary functions.
- Remove unnecessary checks on token transfer.

ITStar10 committed:
Update in progress

Remaining:
	CheckServiceAvailability for user
	User disconnect problem
	And other stuffs

ITStar10 committed:
Merge branch 'feature/24-service-registry' of github.com:verida/blockchain-contracts into feature/24-service-registry

webmaster1225 commented on VDA-Service-Registry/contracts/ServiceRegistry.sol:
```diff
@@ -0,0 +1,466 @@
+        bool isRegistered = registeredIds[identity].contains(serviceId);
+        require(!isRegistered, "Service has already registered!");
+        registeredIds[identity].add(serviceId);
+        AccountInfo storage _account = accountInfos[identity];
```
yes
